### DISCUSSION TITLE
How to scroll through faboptions?

### DISCUSSION THREAD

**Discussion by Pezcraft** [nambicompany/expandable-fab#32] [Q&A]
I have built an expandable fab with 14 faboptions. Is it possible to scroll through those faboptions or create a second column?

**Comment by TacoTheDank** [comment]
I'll just leave [this](https://material.io/components/buttons-floating-action-button#types-of-transitions) here... You should rethink how you have your options set up if you have any more than six.

**Reply by Pezcraft** [reply]
My faboptions consist of 14 different aquarium water tests. The user can select a water test to add it to a dashboard. I think that this library is perfect for this case and so does not violate material guidelines.

**Reply by TacoTheDank** [reply]
Hmm, that actually sounds pretty interesting for a use case. Would you be willing to provide a screenshot?

**Reply by Pezcraft** [reply]
As you see, one can add a water test to display it as a cardview on the dashboard. Currently my faboptions are too small. A second column would be nice or a bigger size with scrolling.
![WhatsApp Image 2021-09-30 at 04 26 19](https://user-images.githubusercontent.com/28001912/135375683-3a2fe3c5-5c93-4c4e-9d8e-1b379ae1a3b7.jpeg)

**Reply by TacoTheDank** [reply]
Hmm. While the rest of the UI looks pretty good in general, people may likely see this as counterintuitive (if it is being published on the Play Store). I think the faboptions look so small because they were not meant to fit so many on screen like that. I'm still of the opinion that it would likely make more sense (and probably look better as well) if you used a different material feature for this, like a scrollable menu or dialog. Ultimately, you're free to use them how you wish, though :)

**Reply by kabumere** [reply]
Hey @Pezcraft,

I tend to agree with @TacoTheDank. Here's my response to a [similar question](https://github.com/nambicompany/expandable-fab/issues/26#issuecomment-899029010) last month. Specifically:

> The ExpandableFab works best in drawing a user's attention and showcasing the most prominent actions of the given screen when there's only a few number of actions available to take. I've found that most times when people start adding 10 or more actions in the ExpandableFab, some of those actions would probably be better suited for a [Navigation Drawer](https://developer.android.com/guide/navigation/navigation-ui#add_a_navigation_drawer) or [Overflow/Options Menu](https://developer.android.com/guide/topics/ui/menus#options-menu).
> 
> If however you simply have a very complex screen that actually does have 20+ relevant actions for a user to take, I still think there are better ways to list those actions for the user. Twenty or more actions in a scrollable ExpandableFab seems like a terrible user experience, in my opinion. Depending on your UI, perhaps try listing these actions in a [Spinner](https://developer.android.com/guide/topics/ui/controls/spinner), [Dialog](https://developer.android.com/guide/topics/ui/dialogs), or scrollable [RecyclerView](https://developer.android.com/guide/topics/ui/layout/recyclerview). There are other premade Views available beyond those that can handle lists of items as well.

So while I'm glad you have interest in the library, I don't think it quite fits your use case. A scrollable list of FabOptions still doesn't seem like good UX (to me, at least).
